pipx/Venv.py

```diff
@@ -107,6 +110,12 @@ def uses_shared_libs(self):
             # always use shared libs when creating a new venv
             return True
 
+    @property
+    def package_metadata(self) -> Dict[str, PackageInfo]:
+        return_dict = self.pipx_metadata.injected_packages.copy()
+        return_dict[self.root.name] = self.pipx_metadata.main_package
+        return return_dict
```

Comment:
I think keeping the distinction between the main package and the injected packages is valuable. 

Recommend making two methods, `injected_package_metadata -> Dict[str, PackageInfo]` and `main_package_metadata -> Dict[str, PackageInfo]`.

Reply:
The dict just takes a package name and returns a PackageInfo object.  All this property is doing is making the main package able to be looked up by name using the same method, instead of having to know that the package you're querying is the main package.  The code got cumbersome before when I had to check if `package` was the main_package every time before querying pipx_metadata.  You can still check pipx_metadata to find main_package apart from the injected packages.

I don't quite understand what you mean here with main_package_metadata, wouldn't it only have one entry?  Why would it be a dict?

This also is how get_venv_metadata_for_package() worked, you just gave it a package name argument, it didn't distinguish between main or injected packages.

Reply:
> I don't quite understand what you mean here with main_package_metadata, wouldn't it only have one entry? Why would it be a dict?

Whoops, I meant `main_package_metadata -> PackageInfo` (copy/paste error).

Is `package_metadata` ever used on anything other than the main package?

Reply:
> Is `package_metadata` ever used on anything other than the main package?

Yes, e.g. `_run_post_install_actions()` is called both by `install()` and `inject()`, and it uses `package_metadata`.

I was also trying to keep `Venv()` more generally useful because it is critical infrastructure for us.  Right now injected packages seem often to be neglected or ignored by the code, but I wanted to put code into place that would allow us to deal with injected packages like first-class citizens in the future if we decided to.

Reply:
Ah okay, yeah this I am fine with this. Sorry for the noise.